packages/pieces/framework/src/lib/piece.ts

```diff
@@ -46,11 +46,12 @@ export class Piece implements PieceBase {
     return this._triggers[triggerName];
   }
 
-  metadata(): PieceMetadata {
+  metadata(type = PieceType.PUBLIC): PieceMetadata {
```

Comment:
why is `type` being passed by the caller?!
Shouldn't the piece contain the `type` information?!

Reply:
Done moved it up